src/main/java/de/tum/in/www1/artemis/service/AssessmentService.java

```diff
@@ -87,35 +88,34 @@ public Result updateAssessmentAfterComplaint(Result originalResult, Exercise exe
         // Save the complaint response
         ComplaintResponse complaintResponse = complaintResponseService.resolveComplaint(assessmentUpdate.getComplaintResponse());
 
-        try {
-            // Store the original result with the complaint
-            Complaint complaint = complaintResponse.getComplaint();
-            complaint.setResultBeforeComplaint(resultService.getOriginalResultAsString(originalResult));
-            complaintRepository.save(complaint);
-        }
-        catch (JsonProcessingException exception) {
-            throw new InternalServerErrorException("Failed to store original result");
-        }
+        // Create a new result which is a copy of the original result.
+        Result newResult = submissionService.createResultAfterComplaintResponse(originalResult.getSubmission(), originalResult, assessmentUpdate.getFeedbacks());
+        newResult.setAssessor(complaintResponse.getReviewer());
+        newResult.setAssessmentType(originalResult.getAssessmentType());
+
+        resultRepository.save(newResult);
+
+        // resultService.addFeedbackToResult(newResult, assessmentUpdate.getFeedbacks(), true);
```

Comment:
I think you forgot to delete the comment here, right @SimonEntholzer ?

Reply:
yep, I forgot this one, I removed it.